--- a/src/llvm/llvm-nm.cc
+++ b/src/llvm/llvm-nm.cc
@@ -52,18 +52,18 @@
 using namespace object;
 
 namespace {
-using namespace llvm::opt; // for HelpHidden in Opts.inc
+using namespace llvm::opt; // for HelpHidden in llvm-nm-opts.inc
 enum ID {
   OPT_INVALID = 0, // This is not an option ID.
 #define OPTION(PREFIX, NAME, ID, KIND, GROUP, ALIAS, ALIASARGS, FLAGS, PARAM,  \
                HELPTEXT, METAVAR, VALUES)                                      \
   OPT_##ID,
-#include "Opts.inc"
+#include "llvm-nm-opts.inc"
 #undef OPTION
 };
 
 #define PREFIX(NAME, VALUE) const char *const NAME[] = VALUE;
-#include "Opts.inc"
+#include "llvm-nm-opts.inc"
 #undef PREFIX
 
 const opt::OptTable::Info InfoTable[] = {
@@ -74,7 +74,7 @@
       METAVAR,     OPT_##ID,  opt::Option::KIND##Class,                        \
       PARAM,       FLAGS,     OPT_##GROUP,                                     \
       OPT_##ALIAS, ALIASARGS, VALUES},
-#include "Opts.inc"
+#include "llvm-nm-opts.inc"
 #undef OPTION
 };
 
@@ -2297,7 +2297,7 @@
   printExportSymbolList(SymbolList);
 }
 
-int main(int argc, char **argv) {
+extern "C" int llvm_nm_main(int argc, char **argv) {
   InitLLVM X(argc, argv);
   BumpPtrAllocator A;
   StringSaver Saver(A);
@@ -2459,4 +2459,5 @@
 
   if (HadError)
     return 1;
+  return 0;
 }
